### ISSUE TITLE
`Save image` dialog box opens if text is pasted in ReText

### ISSUE THREAD

**Issue by holzkohlengrill** [retext-project/retext#385]
When I paste text into ReText sometimes it opens the `save image` dialog box instead of pasting the text. To resolve this I have to paste it first in another plain text editor (like Notepad++, gedit, ...) to paste it in ReText. This happens for me since several major ReText versions.

I still can't give you a totally reproducable scenario for this.

Data when I copy text from ReText:

![image](https://user-images.githubusercontent.com/12732886/45421089-47bc4e80-b68c-11e8-810c-ce68f9ae8155.png)

Data when I copy (formatted) text from MS OneNote (this does only sometime occur using OneNote but also with different applications):

![image](https://user-images.githubusercontent.com/12732886/45421244-c1ecd300-b68c-11e8-81ab-72e35f1c98ed.png)

Text pasted in Notepad++ and then copied again:

![image](https://user-images.githubusercontent.com/12732886/45421425-42133880-b68d-11e8-97f2-ecabd2723b03.png)

==============

ReText: 7.1.0 (Git)
PyMarkups: 3.0.0
Qt: 5.10.1
Python: 3.7.0b5

**Comment by mitya57** [comment]
We are not doing any magic here, and relying on a standard Qt method, [QMimeData::hasImage()](http://doc.qt.io/qt-5/qmimedata.html#hasImage). So it may indicate a bug somewhere in Qt.

Can you try to actually save the image, and check its contents after saving? Maybe this will give us some clue.

**Comment by holzkohlengrill** [comment]
> Can you try to actually save the image, and check its contents after saving? Maybe this will give us some clue.

Than python crashes :smile: 

```
Connected to pydev debugger (build 181.4892.52)
Using configuration file: C:/Users/holzkohlengrill/AppData/Roaming/ReText project/ReText.ini
Traceback (most recent call last):
  File "C:\Users\holzkohlengrill\AppData\Local\Programs\retext\ReText\editor.py", line 403, in insertFromMimeData
    fileName, link = self.getImageFilenameAndLink()
  File "C:\Users\holzkohlengrill\AppData\Local\Programs\retext\ReText\editor.py", line 393, in getImageFilenameAndLink
    link = os.path.relpath(chosenFileName, saveDir)
  File "C:\Program Files (x86)\Python37-32\lib\ntpath.py", line 564, in relpath
    path_drive, start_drive))
ValueError: path is on mount 'C:', start on mount 'E:'
```
What is strange is that regardless on which drive (c: or e:) I am trying to save the image I get `ValueError: path is on mount 'C:', start on mount 'E:'`
Just figured it out: This was because some documents were opened on drive e: (that's where I pasted my clipboard into).

Otherwise (when on drive c:; open documents + saved image) `![image0001](image0001.png)` gets pasted into ReText and a proper image (.png) was saved. Proof:

![image0001](https://user-images.githubusercontent.com/12732886/45688634-35418980-bb52-11e8-977d-28a9e959dbc3.png)



**My proposal would be to paste it as text by default but to add another option in the context menu like "Paste as image" if an image is available in the clipboard (otherwise greyed out).**

I did not take a deep look into the drive issue but I expect this is probably a bug which should be fixed.

**Comment by mitya57** [comment]
This should be fixed now, can you please test?

- ValueError fixed in 4c24ee3bc2f182cbfb8391350d76cc584137b345;
- Paste Image is moved to a separate action in a19d6e63ce27578e3739436363bfab7b14c1924e.

**Comment by holzkohlengrill** [comment]
Works now smoothly. Tested markdown text:

    ![image0001](C:/Users/holzkohlengrill/Downloads/0-temp/image0001.png)


    ```
    docker image ls
    ```

    * List Docker containers (running, all, all in quiet mode)
    ```
    docker container ls
    docker container ls --all
    docker container ls -aq
    ```

    ## Docker File
    * `Dockerfile` defines what goes on in the environment inside your container.
    * Map networking interfaces and disk drives virtualized inside this environment (ports) to the outside world


    ![image0001](E:/image0001.png)
